libs/wingsdk/src/cloud/api.ts

```diff
@@ -360,6 +360,54 @@ export abstract class Api extends Resource {
     };
   }
 
+  /**
+   * Checks if two given paths are ambiguous.
+   * @param pathA
+   * @param pathB
+   * @returns A boolean value indicating if provided paths are ambiguous.
+   * @internal
+   */
+  protected _arePathsAmbiguous(pathA: string, pathB: string): boolean {
+    const partsA = pathA.split("/");
+    const partsB = pathB.split("/");
+
+    if (partsA.length !== partsB.length) {
+      return false;
+    }
+
+    for (let i = 0; i < partsA.length; i++) {
+      const partA = partsA[i];
+      const partB = partsB[i];
+
+      if (
+        partA !== partB &&
+        !partA.match(/^{.+?}$/) &&
+        !partB.match(/^{.+?}$/)
+      ) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Checks if provided path and method are ambigous with paths and methods already defined in the api spec.
+   * @param path Path to be checked
+   * @param method HTTP method
+   * @returns A boolean value indicating if provided path and method are ambiguous.
+   * @internal
+   */
+  protected _isPathAndMethodAmbiguous(path: string, method: string): boolean {
+    const existingPaths = Object.keys(this.apiSpec.paths);
```

Comment:
are those of the same method?

Reply:
`existingPaths` contains paths of all HTTP methods, in the next lines these are narrowed down to a single method that will be added